src/test/java/com/amazon/ion/conformance/expectations.kt

```diff
@@ -51,22 +50,25 @@ fun TestCaseSupport.assertSignals(sexp: SeqElement, r: IonReader) {
  * if an error is _not_ encountered, we have some useful information for debugging the test failure.
  */
 private fun IonReader.walk(): List<String> {
-    val events = mutableListOf("START")
+    val events = mutableListOf<String>()
     fun recordEvent(eventType: String = type.toString(), value: Any? = "") {
         events.add("[$eventType] $value")
     }
+    recordEvent("START")
 
     while (true) {
         next()
         val currentType = type
-            ?: try {
+        if (type == null) {
+            if (depth > 0) {
                 stepOut()
                 recordEvent("STEP-OUT")
                 continue
-            } catch (e: IllegalStateException) {
+            } else {
```

Comment:
🗺️ I don't know why this was catching an exception here. It should be allowing it to propagate up to fail the test.